judge/admin/problem.py

```diff
@@ -248,3 +248,17 @@ def construct_change_message(self, request, form, *args, **kwargs):
         if form.cleaned_data.get('change_message'):
             return form.cleaned_data['change_message']
         return super(ProblemAdmin, self).construct_change_message(request, form, *args, **kwargs)
+
+
+class ProblemPointsVoteAdmin(admin.ModelAdmin):
+    list_display = ('points', 'voter', 'problem', 'vote_time')
+    search_fields = ('voter', 'problem')
+    readonly_fields = ('voter', 'problem', 'vote_time')
+
+    def has_change_permission(self, request, obj=None):
```

Comment:
This code itself is fine, but you must define `get_queryset`. As it stands, a staff user with problem setter perms can see every. single. vote on the site.

Reply:
I used the same `get_queryset()` as problemAdmin:
```py
    def get_queryset(self, request):
        return Problem.get_editable_problems(request.user).prefetch_related('authors__user').distinct()        
```

Reply:
No, this doesn't work even a little bit. You are returning `Problem` objects. Do this:

```py
    def get_queryset(self, request):
        return ProblemPointsVote.objects.filter(problem__in=Problem.get_editable_problems(request.user)).all()
```

(Tested).

Reply:
Nit: the `.all()` is not needed.

Reply:
Ah, wasn't sure. Remove it when you update this.